frame/recovery/src/benchmarking.rs

```diff
@@ -0,0 +1,384 @@
+	}: _(
+		RawOrigin::Signed(caller),
+		recovered_account,
+		Box::new(call)
+	)
+
+	set_recovered {
+		let lost: T::AccountId = whitelisted_caller();
+		let rescuer: T::AccountId = whitelisted_caller();
+	}: _(
+		RawOrigin::Root,
+		lost.clone(),
+		rescuer.clone()
+	) verify {
+		assert_last_event::<T>(
+			Event::AccountRecovered {
+				lost_account: lost,
+				rescuer_account: rescuer,
+			}.into()
+		);
+	}
+
+	create_recovery {
+		let n in 1 .. T::MaxFriends::get();
+
+		let caller: T::AccountId = whitelisted_caller();
+		T::Currency::make_free_balance_be(&caller, BalanceOf::<T>::max_value());
+
+		// Create friends
+		let friends = generate_friends::<T>(n);
+	}: _(
+		RawOrigin::Signed(caller.clone()),
+		friends,
+		n as u16,
+		42u32.into()
+	) verify {
+		assert_last_event::<T>(Event::RecoveryCreated { account: caller }.into());
+	}
+
+	initiate_recovery {
+		let caller: T::AccountId = whitelisted_caller();
+		T::Currency::make_free_balance_be(&caller, BalanceOf::<T>::max_value());
+
+		let lost_account: T::AccountId = account("lost_account", 0, SEED);
+
+		insert_recovery_account::<T>(&caller, &lost_account);
+	}: _(
+		RawOrigin::Signed(caller.clone()),
+		lost_account.clone()
+	) verify {
+		assert_last_event::<T>(
+			Event::RecoveryInitiated {
+				lost_account: lost_account,
+				rescuer_account: caller,
+			}.into()
+		);
+	}
+
+	vouch_recovery {
+		let n in 1 .. T::MaxFriends::get();
+
+		let caller: T::AccountId = whitelisted_caller();
+		let lost_account: T::AccountId = account("lost_account", 0, SEED);
+		let rescuer_account: T::AccountId = account("rescuer_account", 0, SEED);
+
+		// Create friends
+		let friends = add_caller_and_generate_friends::<T>(caller.clone(), n);
+		let bounded_friends: FriendsOf<T> = friends.try_into().unwrap();
+
+		let delay_period = block_number::<T>();
+
+		// Get deposit for recovery
+		let total_deposit = get_total_deposit::<T>(&bounded_friends).unwrap();
+
+		let recovery_config = RecoveryConfig {
+			delay_period,
+			deposit: total_deposit.clone(),
+			friends: bounded_friends.clone(),
+			threshold: n as u16,
+		};
+
+		// Create the recovery config storage item
+		<Recoverable<T>>::insert(&lost_account, recovery_config.clone());
+
+		// Reserve deposit for recovery
+		T::Currency::reserve(&caller, total_deposit).unwrap();
+
+		// Create an active recovery status
+		let recovery_status = ActiveRecovery {
+			created: block_number::<T>(),
+			deposit: total_deposit,
+			friends: generate_friends::<T>(n - 1).try_into().unwrap(),
+		};
+
+		// Create the active recovery storage item
+		<ActiveRecoveries<T>>::insert(&lost_account, &rescuer_account, recovery_status);
+
+	}: _(
+		RawOrigin::Signed(caller.clone()),
+		lost_account.clone(),
+		rescuer_account.clone()
+	) verify {
+		assert_last_event::<T>(
+			Event::RecoveryVouched {
+				lost_account: lost_account,
+				rescuer_account: rescuer_account,
+				sender: caller,
+			}.into()
+		);
+	}
+
+	claim_recovery {
```

Comment:
not a big deal, but a lot of the code in `create_recovery`, `vouch_recovery`, `claim_recovery` are common, and build on top of one another. You can have something like 

```rust
enum RecoveryStage {
  Create,
  Vouche(u16),
  Claim,
}

fn create_recovery_scenario(stage: RecoveryStage) {}
```

Reply:
Hmm...okay, I can play around with this and make things cleaner.